test/hotspot/jtreg/compiler/vectorapi/TestVectorShuffleIotaShort.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2021, Huawei Technologies Co. Ltd. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package compiler.vectorapi;
+
+import jdk.incubator.vector.ShortVector;
+import jdk.incubator.vector.VectorSpecies;
+import jdk.incubator.vector.VectorShuffle;
+
+/*
+ * @test
+ * @bug 8265956
+ * @modules jdk.incubator.vector
+ * @run main/othervm compiler.vectorapi.TestVectorShuffleIotaShort
+ */
+
+public class TestVectorShuffleIotaShort {
+    static final VectorSpecies<Short> SPECIESs = ShortVector.SPECIES_128;
+
+    static final int INVOC_COUNT = 50000;
+
+    static short[] as = {87, 65, 78, 71, 72, 69, 82, 69};
+
+    public static void testShuffleS() {
```

Comment:
IMO, this test is not enough since the assembly code is not easy to get reviewed.

So I suggest to improve your test:
 1) Test as many match-patterns you added as possible
 2) Verify the results in your tests

Thanks.

Reply:
Thank you for your review.
1 There is only one match-pattern here .
2 I will add checking code here.

Reply:
@DamonFool I will add the other vector length here. Thank you for your review!